Test/Unit/Helper/OrderTest.php

```diff
@@ -1064,27 +1065,28 @@ private function dispatchPostCheckoutEventsSetUp()
      */
     public function dispatchPostCheckoutEvents()
     {
-        list($orderMock, $quoteMock) = $this->dispatchPostCheckoutEventsSetUp();
-        $quoteMock->expects(self::once())->method('getBoltReservedOrderId')->willReturn(true);
-        $quoteMock->expects(self::once())->method('setInventoryProcessed')->with(true);
+        list($orderMock, $immutableQuoteMock) = $this->dispatchPostCheckoutEventsSetUp();
+        $immutableQuoteMock->expects(self::once())->method('getBoltReservedOrderId')->willReturn(true);
+        $immutableQuoteMock->expects(self::once())->method('setInventoryProcessed')->with(true);
 
         $orderMock->expects(self::once())->method('getAppliedRuleIds')->willReturn(null);
         $orderMock->expects(self::once())->method('setAppliedRuleIds')->with('');
+        $orderMock->expects(self::once())->method('setQuoteId')->with($immutableQuoteMock->getId())->willReturnSelf();
```

Comment:
@danac-gs 
Expectation added here to the immutable quote ID replacing the the session quote ID on the order as described above.